Jenkinsfile

```diff
@@ -66,20 +66,27 @@ def job = {
         downstreamBuildsStepName: {
             echo "Building cp-downstream-builds"
             if (config.isPrJob) {
-                def muckrakeBranch = kafkaMuckrakeVersionMap[env.CHANGE_TARGET]
-                def forkRepo = "${env.CHANGE_FORK ?: "confluentinc"}/kafka.git"
-                def forkBranch = env.CHANGE_BRANCH
-                echo "Schedule test-cp-downstream-builds with :"
-                echo "Muckrake branch : ${muckrakeBranch}"
-                echo "PR fork repo : ${forkRepo}"
-                echo "PR fork branch : ${forkBranch}"
-                buildResult = build job: 'test-cp-downstream-builds', parameters: [
-                        [$class: 'StringParameterValue', name: 'BRANCH', value: muckrakeBranch],
-                        [$class: 'StringParameterValue', name: 'TEST_PATH', value: "muckrake/tests/dummy_test.py"],
-                        [$class: 'StringParameterValue', name: 'KAFKA_REPO', value: forkRepo],
-                        [$class: 'StringParameterValue', name: 'KAFKA_BRANCH', value: forkBranch]],
-                        propagate: true, wait: true
-                return ", downstream build result: " + buildResult.getResult();
+                try {
+                    def muckrakeBranch = kafkaMuckrakeVersionMap[env.CHANGE_TARGET]
+                    def forkRepo = "${env.CHANGE_FORK ?: "confluentinc"}/kafka.git"
+                    def forkBranch = env.CHANGE_BRANCH
+                    echo "Schedule test-cp-downstream-builds with :"
+                    echo "Muckrake branch : ${muckrakeBranch}"
+                    echo "PR fork repo : ${forkRepo}"
+                    echo "PR fork branch : ${forkBranch}"
+                    buildResult = build job: 'test-cp-downstream-builds', parameters: [
+                            [$class: 'StringParameterValue', name: 'BRANCH', value: muckrakeBranch],
+                            [$class: 'StringParameterValue', name: 'TEST_PATH', value: "muckrake/tests/dummy_test.py"],
+                            [$class: 'StringParameterValue', name: 'KAFKA_REPO', value: forkRepo],
+                            [$class: 'StringParameterValue', name: 'KAFKA_BRANCH', value: forkBranch]],
+                            propagate: true, wait: true
+                    downstreamBuildFailureOutput = "cp-downstream-builds result: " + buildResult.getResult();
+                    return downstreamBuildFailureOutput
+                } catch (ignored) {
+                    downstreamBuildFailureOutput = "cp-downstream-builds result: " + e.getMessage()
+                    currentBuild.result = 'UNSTABLE'
```

Comment:
Would this potentially make a red build yellow if the main build did not compile?

Reply:
To clarify, the question is what currentBuild affects.

Reply:
I tested by adding an error in the compile stage and it did mark it as red, however I can't say for sure. I was considering trying a change like:

```
if (currentBuild.result.equals('FAILED')) {
     currentBuild.result = 'UNSTABLE'
}
```

What do you think? It seems safer if it works but it'll require more testing of failures in all the phases.

Reply:
That code is supposed to be `if (!currentBuild.result.equals('FAILURE'))`

Reply:
If you tested a scenario where this PR has a compiler error and by adding an exception to the downstream try block and it worked, then I think we don't need to add more logic.

Reply:
I have tested this even more thoroughly, including adding some delays in different places, and I am pretty confident it works. I made one more change here that I think is useful https://github.com/confluentinc/kafka/commit/a79ebb8ee197917745f36d3d7a95f9aeeed030d5, but I think we can consider merging it once the latest test run finishes.

Reply:
Do we archive it in Jenkins or also in the snapshots repo?

Reply:
We archive it in jenkins in the same way we archive stdout for the failing unit tests. It's not placed in artifactory which uses gradle instead.